SoftwareDesign.md

```diff
@@ -0,0 +1,8 @@
+## Software Design
+
+  -My favorite software design pattern that I learned this week is Lazy evaluation. 
+  This was probably my favorite becasue it was easy for me to understand - and made alot of sense to me(values don't need to be computed if they're not going to be used.) This design pattern would be useful in situations where you do not want to use alot of memory and you care about the overall runtime.
+  
+ -A software design anti-pattern that I've run into is "copy and paste programming",
+ I used this method of copying and modifying existing code alot. My thought pattern for doing so was "why re-invent the wheel?".
+ I did this alot in the first few weeks of Lambda School- thinking if I had already done a similar project or if I saw something similar on "Stack OverFlow" then it wouldn't hurt to copy and paste it and then change a few things around. The major problem with this was it wouldnt always work. The tasks may have looked similar but they really weren't. I realized this early on- and also realized that while I may not want to re-invent the wheel-- I do want to learn as much as possible. My solution to this anti pattern is to carefully read and step through code that looks simliar to something I am working on. Sometimes I find out what I am doing wrong in my existing code, and other times I find out that the code I am referancing doesn't even help complete the task I am trying to complete.

```

Comment:
I agree that copy and pasting without understanding the problem behind your code or what the copied code does can be detrimental. That being said, copying and pasting code can be helpful at certain times, but it's important that you take time to really understand what that code is doing and be able to do any necessary refactors to suit your needs.